### PR TITLE
Fix bitmap decode OutOfMemoryError and make image display more clearly

### PR DESCRIPTION
**Pull request by start141**

### CODE REVIEW

**Comment by start141** [comment]
before and after
qq20160218-0 2x

File: ucrop/src/main/java/com/yalantis/ucrop/util/BitmapLoadUtils.java [hunk 1]
```diff
@@ -82,10 +95,10 @@ public static int calculateInSampleSize(@NonNull BitmapFactory.Options options,
         int inSampleSize = 1;
 
         if (height > reqHeight || width > reqWidth) {
-            // Calculate the largest inSampleSize value that is a power of 2 and keeps both
+            // Calculate the largest inSampleSize value and keeps both
             // height and width lower or equal to the requested height and width.
             while ((height / inSampleSize) > reqHeight || (width / inSampleSize) > reqWidth) {
-                inSampleSize *= 2;
+                inSampleSize++;
```
Comment (shliama) [inline ucrop/src/main/java/com/yalantis/ucrop/util/BitmapLoadUtils.java]: From developer.android.com
If set to a value > 1, requests the decoder to subsample the original image, returning a smaller image to save memory. The sample size is the number of pixels in either dimension that correspond to a single pixel in the decoded bitmap. For example, inSampleSize == 4 returns an image that is 1/4 the width/height of the original, and 1/16 the number of pixels. Any value <= 1 is treated the same as 1. **Note: the decoder uses a final value based on powers of 2, any other value will be rounded down to the nearest power of 2.**

Comment (start141) [inline ucrop/src/main/java/com/yalantis/ucrop/util/BitmapLoadUtils.java]: Yeah! But you limited request width and height already. I think the inSampleSize may can smaller!

**Comment by start141** [comment]
What time can fix the OutOfMemoryError? thanks!

**Comment by shliama** [comment]
So, as I thought:
Note: the decoder uses a final value based on powers of 2, **any other value will be rounded down to the nearest power of 2.**

I took a photo with size 5690x4268 pixels. Max bitmap size was 2094 pixels as for my Nexus 5x. When `inSampleSize` was calculated with your correction - it was 3. (5690 / 3 < 2094 && 4268 / 3 < 2094).
But while bitmap was decoded, 3 was rounded down to the value of 2. So returned bitmap has the size of 2845x2134 pixels - which isn't the logic I want.
While this size may be ok for viewing/zooming an image - it may result in OOM when cropping it.

Therefor, if you want to increase bitmap size - just use `Options#setMaxBitmapSize()` method.

However, I'll add check for OOM together with extracting image decoding logic from UI thread soon.

**Comment by start141** [comment]
thanks.

**Comment by shliama** [comment]
OOM handling added in 1.3.0

**Pull request closed without merging**